tests/test_regressor/test_residuals.py

```diff
@@ -323,3 +351,19 @@ def test_score(self):
         assert score == pytest.approx(0.9999888484, rel=1e-4)
         assert visualizer.train_score_ == pytest.approx(0.9999906, rel=1e-4)
         assert visualizer.test_score_ == score
+
+    def test_alpha_param(self):
+        """
+        Test that the user can supply an alpha param on instantiation
+        """
+        # Instantiate a prediction error plot, provide custom alpha
+        visualizer = ResidualsPlot(
+            Ridge(random_state=8893), alpha=0.3
+        )
+
+        # Test param gets set correctly
+        assert visualizer.alpha == 0.3
+
+        # TODO: mock ax and test alpha is passed to pyplot scatterplot
+        # not sure how to do this without triggering the Matplotlib error:
+        # "ValueError: Axes instance argument was not found in a figure"
```

Comment:
Unfortunately setting `hist=False` does not correct the error. Here are the test results:

```
    def test_alpha_param(self):
        """
            Test that the user can supply an alpha param on instantiation
            """
        # Instantiate a prediction error plot, provide custom alpha
        visualizer = ResidualsPlot(
            Ridge(random_state=8893), alpha=0.3, hist=False
        )
    
        # Test param gets set correctly
        assert visualizer.alpha == 0.3
    
        # TODO: mock ax and test alpha is passed to pyplot scatterplot
        # not sure how to do this without triggering the Matplotlib error:
        # "ValueError: Axes instance argument was not found in a figure"
        # Mock ax and fit the visualizer
        visualizer.ax = mock.MagicMock(autospec=True)
>       visualizer.fit(self.data.X.train, self.data.y.train)

test_residuals.py:372: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../../yellowbrick/regressor/residuals.py:463: in fit
    self.score(X, y, train=True)
../../yellowbrick/regressor/residuals.py:498: in score
    self.draw(y_pred, scores, train=train)
../../yellowbrick/regressor/residuals.py:553: in draw
    plt.sca(self.ax)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

ax = <MagicMock id='4632511600'>

    def sca(ax):
        """
        Set the current Axes instance to *ax*.
    
        The current Figure is updated to the parent of *ax*.
        """
        managers = _pylab_helpers.Gcf.get_all_fig_managers()
        for m in managers:
            if ax in m.canvas.figure.axes:
                _pylab_helpers.Gcf.set_active(m)
                m.canvas.figure.sca(ax)
                return
>       raise ValueError("Axes instance argument was not found in a figure.")
E       ValueError: Axes instance argument was not found in a figure.

/Library/Frameworks/Python.framework/Versions/3.6/lib/python3.6/site-packages/matplotlib/pyplot.py:963: ValueError

```

Reply:
Ah, that traceback helps a lot. So in `draw()` and possibly in `finalize()`I call `plt.sca` (set current axes) to make sure that `viz.ax` is the global axes instead of `viz.hax` -- I guess that function doesn't like getting a mock! Can you patch the sca function? 

```python
@mock.patch('yellowbrick.regressor.residuals.plt.sca', autospec=True) 
def test_alpha_param(self):
    ... 
```

Patching is a bit of a trial and error thing, especially since the order of imports matters. If that doesn't work, you could also try `@mock.patch('matplotlib.pyplot.sca')` ... and if that doesn't work, then we can try other things.

Reply:
Oh sorry, this is actually a fairly simple fix*:

```python
@mock.patch('yellowbrick.regressor.residuals.plt.sca', autospec=True) 
def test_alpha_param(self, mock_sca):
    ... 
```

I forgot the second argument to the test function that is passed in by the decorator. 

* though the unittest dependency might screw us up ... 

However, if you want to simply mark this as `xfail` -- that's fine, we can fix it later.

Reply:
Ok, making one last attempt; the test passes on my machine with python 3.6.

Reply:
🤞